app/services/title_builder.rb

```diff
@@ -4,38 +4,130 @@ class TitleBuilder
   # @param [Array<Cocina::Models::Title>] titles
   # @returns [String] the title value for Solr
   def self.build(titles)
-    title = titles.first
+    title = primary_title(titles) || first_untyped_title(titles) || titles.first
     if title.value
```

Comment:
Can you pull the variable assignment out of the conditional, e.g:
```ruby
my_title = if title.value
  title.value
elsif title.structuredValue
  title_from_structured_values(...)
else
...
end
```

Reply:
good idea; done.